packages/sanity/src/core/hooks/useUserListWithPermissions.ts

```diff
@@ -1,39 +1,71 @@
 /* eslint-disable max-nested-callbacks */
 import {type SanityDocument} from '@sanity/client'
+import {type User} from '@sanity/types'
 import {sortBy} from 'lodash'
 import {useEffect, useMemo, useState} from 'react'
 import {concat, forkJoin, map, mergeMap, type Observable, of, switchMap} from 'rxjs'
+
 import {
-  DEFAULT_STUDIO_CLIENT_OPTIONS,
+  type DocumentValuePermission,
   grantsPermissionOn,
   type ProjectData,
-  useClient,
   useProjectStore,
   useUserStore,
-} from 'sanity'
+} from '../store'
+import {DEFAULT_STUDIO_CLIENT_OPTIONS} from '../studioClient'
+import {useClient} from './useClient'
+
+type Loadable<T> = {
+  data: T | null
+  error: Error | null
+  loading: boolean
+}
 
-import {type Loadable, type MentionOptionsHookValue, type MentionOptionUser} from '../types'
+/**
+ * @beta
+ * @hidden
+ */
+export type UserListWithPermissionsHookValue = Loadable<UserWithPermission[]>
+
+/**
+ * @beta
+ * @hidden
+ */
+export interface UserWithPermission extends User {
+  granted: boolean
+}
 
-const INITIAL_STATE: MentionOptionsHookValue = {
+const INITIAL_STATE: UserListWithPermissionsHookValue = {
   data: [],
   error: null,
   loading: true,
 }
 
-export interface MentionHookOptions {
+/**
+ * @beta
+ */
+export interface UserListWithPermissionsOptions {
   documentValue: SanityDocument | null
+  permission: DocumentValuePermission
 }
 
 let cachedSystemGroups: [] | null = null
 
-export function useMentionOptions(opts: MentionHookOptions): MentionOptionsHookValue {
-  const {documentValue} = opts
+/**
+ * @beta
+ * returns a list of users with the specified permission on the document.
+ * If not document is provided it will return all as `granted: true`
+ */
```

Comment:
Nitpick – small update suggestion for this TSdoc comment:
- Tiny grammar fix ("no" instead of "not")
- Capitalize "R" in "returns"